### PR TITLE
chore(test): Remove globals from mocks.

### PR DESCRIPTION
**Pull request by kinyoklion**
This removes the global platform, logger, clientContext, and hash.

I've not investigated fixing the event processor mock or streaming processor mock yet.

It also removes the global setupMocks requirement for the mocks package.

This PR, aside from the mocks package, should only change tests.

The mocks package is not published and can only be used within the repo, so we don't need to do a release.

### CODE REVIEW

File: packages/sdk/react-native/src/fromExternal/react-native-sse/EventSource.test.ts [hunk 1]
```diff
@@ -1,8 +1,14 @@
 import { type EventName } from '@launchdarkly/js-client-sdk-common';
-import { logger } from '@launchdarkly/private-js-mocks';
+import { createLogger } from '@launchdarkly/private-js-mocks';
 
 import EventSource, { backoff, jitter } from './EventSource';
 
+let logger: ReturnType<typeof createLogger>;
```
Comment (kinyoklion) [inline packages/sdk/react-native/src/fromExternal/react-native-sse/EventSource.test.ts:6]: This lets the createX functions have inferred typed from typescript, so you can tell they are mocks and call the mock functions, while also allowing them to still be typed and provide intellisense.

Here logger is the return type of the type of the createLogger function.

File: packages/sdk/server-node/__tests__/LDClientNode.bucketing.test.ts [hunk 2]
```diff
@@ -3,11 +3,12 @@
  * These tests exercise the hashing from the node platform.
  */
 import { integrations } from '@launchdarkly/js-server-sdk-common';
-import { TestData } from '@launchdarkly/js-server-sdk-common/dist/integrations';
```
Comment (kinyoklion) [inline packages/sdk/server-node/__tests__/LDClientNode.bucketing.test.ts:6]: Fixed a few of the imports from dist. Generally speaking we do not want to do this.

File: packages/shared/sdk-client/src/LDClientImpl.events.test.ts [hunk 3]
```diff
@@ -18,6 +17,17 @@ import * as mockResponseJson from './evaluation/mockResponse.json';
 import LDClientImpl from './LDClientImpl';
 import { Flags } from './types';
 
+type InputCustomEvent = internal.InputCustomEvent;
+type InputIdentifyEvent = internal.InputIdentifyEvent;
+
+let mockPlatform: ReturnType<typeof createBasicPlatform>;
```
Comment (kinyoklion) [inline packages/shared/sdk-client/src/LDClientImpl.events.test.ts:23]: I did this rote conversion for  most of the tests. Most test suites could probably be further improved by reducing the scope of the mocks to specific fixtures (describe) or tests.

File: packages/shared/sdk-client/src/context/addAutoEnv.test.ts [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
+  type LDContext,
+  LDMultiKindContext,
+  LDUser,
+} from '@launchdarkly/js-sdk-common';
+import { createBasicPlatform, createLogger } from '@launchdarkly/private-js-mocks';
 
 import Configuration from '../configuration';
 import { addApplicationInfo, addAutoEnv, addDeviceInfo, toMulti } from './addAutoEnv';
 
+let mockPlatform: ReturnType<typeof createBasicPlatform>;
```
Comment (kinyoklion) [inline packages/shared/sdk-client/src/context/addAutoEnv.test.ts:13]: The rename to mock was just to aid in progress being able to tell which files were not updated. Slowly removing all 'basicPlatform' references.

File: packages/shared/common/__tests__/internal/events/EventProcessor.test.ts [hunk 5]
```diff
@@ -1,13 +1,42 @@
-import { clientContext, ContextDeduplicator } from '@launchdarkly/private-js-mocks';
```
Comment (kinyoklion) [inline packages/shared/common/__tests__/internal/events/EventProcessor.test.ts:1]: The global clientContext mock was only used in a few tests, so I removed the functionality entirely. The test can just specify its clientContext.

**Review by tanderson-ld — APPROVED** [review]

**Pull request merged**